### PR TITLE
Add HTTP/2 transport integration tests

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

The http/2 transports are only tested via unit tests.

Modifications:

- Add a number of integration tests for the HTTP/2 transports

Result:

Better test coverage

### CODE REVIEW

File: Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift [hunk 1]
```diff
@@ -0,0 +1,1377 @@
...
+      $0.numberOfMessages = 1
+      $0.messageParams = .with {
+        $0.content = 42
+        $0.size = 1024
+      }
+    }
+
+    var options = CallOptions.defaults
+    options.compression = client
+    // TODO: pass in options to request
```
Comment (gjcairo) [inline Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift:735]: This is still pending here and in the other tests.

File: Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift [hunk 2]
```diff
@@ -0,0 +1,1377 @@
...
+      var options = CallOptions.defaults
+      options.timeout = .seconds(10)
+      try await control.clientStream(request: request, options: options) { response in
+        let timeout = Array(response.metadata["echo-grpc-timeout"])
+        XCTAssertEqual(timeout.count, 1)
+      }
+    }
+  }
+
+  func testServerStreamTimeoutPropagatedToServer() async throws {
```
Comment (gjcairo) [inline Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift:1174]: NIt
```suggestion
  func testServerStreamingTimeoutPropagatedToServer() async throws {
```

File: Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift
Comment (gjcairo) [inline Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift:1]: Can we add a test that checks the request is rejected if the path doesn't exist on the server (i.e. the service or method aren't handled/are unknown)

File: Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift [hunk 3]
```diff
@@ -0,0 +1,1377 @@
...
+        }
+
+        client.close()
+        group.cancelAll()
+      }
+    }
+  }
+
+  func forEachClientAndHTTPStatusServer(
+    _ kind: [Transport.Kind] = [.posix],
```
[the remark this answers is not in the document]
Comment (glbrntt) [reply Tests/GRPCHTTP2TransportTests/HTTP2TransportTests.swift:104]: There's no NIOTS client at the moment, it'll be added to the defaults.

**Review by gjcairo — APPROVED** [review]

**Pull request merged**